src/main/java/jenkins/plugins/logstash/configuration/LogstashIndexer.java

```diff
@@ -0,0 +1,172 @@
+package jenkins.plugins.logstash.configuration;
+
+import javax.annotation.Nonnull;
+
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.StaplerRequest;
+
+import hudson.DescriptorExtensionList;
+import hudson.ExtensionPoint;
+import hudson.model.AbstractDescribableImpl;
+import hudson.model.Descriptor;
+import hudson.model.ReconfigurableDescribable;
+import hudson.model.Descriptor.FormException;
+import hudson.util.FormValidation;
+import jenkins.model.Jenkins;
+import jenkins.plugins.logstash.Messages;
+import jenkins.plugins.logstash.persistence.AbstractLogstashIndexerDao;
+import net.sf.json.JSONObject;
+
+/**
+ * Extension point for logstash indexers.
+ * This extension point provides the configuration for the indexer. You also have to implement the actual
+ * indexer in a separate class extending {@link AbstractLogstashIndexerDao}.
+ *
+ * @param <T> The class implementing the push to the indexer
+ */
+public abstract class LogstashIndexer<T extends AbstractLogstashIndexerDao>
+    extends AbstractDescribableImpl<LogstashIndexer<?>>
+    implements ExtensionPoint, ReconfigurableDescribable<LogstashIndexer<?>>
+{
+  protected String host;
+  protected int port;
```

Comment:
On a different note: maybe we should push the host and port up?
`MemoryDao` is one obvious example that doesn't use those, but I think it might be cleaner code in general.
We could have another helper abstract class to avoid copying this for all the indexer types

Reply:
If we replace for elastic search the host and port with a url this might make sense

Reply:
done